docfx_yaml/extension.py

```diff
@@ -732,6 +732,52 @@ def extract_product_name(name):
     return product_name
 
 
+def _extract_type_name(annotation: Any) -> str:
```

Comment:
Are annotations and type hints the same? If not, I'd rename `annotation` to `type_hint`, or maybe `argument_annotations`.  Feel free to Ack if this breaks your current naming conventions.

Reply:
I don't think it's quite the same - annotations may hint at the type hint, however it's more like the object that includes the type hint. Probably best to stick to `annotations` from https://docs.python.org/3/library/inspect.html#inspect.get_annotations.